compiler/rustc_hir_typeck/src/closure.rs

```diff
@@ -734,10 +734,11 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
 
         // async fn that have opaque types in their return type need to redo the conversion to inference variables
         // as they fetch the still opaque version from the signature.
+        let body_def_id = self.tcx.hir().local_def_id(body_id);
```

Comment:
please propagate the change outwards, changing the argument `body_id` to `LocalDefId`